library/Backends/llvm/src/Juvix/Backends/LLVM/Compilation.hs

```diff
@@ -49,13 +50,24 @@ compileProgram t =
       show err
         |> Feedback.fail
 
-register ::
-  Types.Define m => ErasedAnn.AnnTerm PrimTy RawPrimVal -> m LLVM.Operand
-register t = do
+registerPreProcessed ::
+  Types.Define m => Types.Annotated Types.TermLLVM -> m LLVM.Operand
+registerPreProcessed t = do
   Closure.register
   Block.defineMalloc
   Block.defineFree
-  mkMain (preProcess t)
+  mkMain t
+
+register ::
+  Types.Define m => ErasedAnn.AnnTerm PrimTy RawPrimVal -> m LLVM.Operand
+register = registerPreProcessed . preProcess
+
+termLLVMToModule ::
+  Types.Annotated Types.TermLLVM -> Either Types.Errors LLVM.Module
+termLLVMToModule t =
+  case Block.runEnvState (registerPreProcessed t) mempty of
+    (Right _, state) -> Right $ state |> Types.moduleAST
+    (Left err, _) -> Left err
```

Comment:
I'm guessing this is the new runner for `llvm-hs-9.0.1`, seems you don't run the `preProcess`at all?

Reply:
Right -- if you want to run with the jit, you call that, and then you can execute the module in some way along the lines of what `runIntModule` does (in `library/Backends/llvm/test/Test/Codegen.hs`).